src/auth/awsclient.ts

```diff
@@ -168,7 +168,9 @@ export class AwsClient extends BaseExternalAccountClient {
     // Initialize AWS request signer if not already initialized.
     if (!this.awsRequestSigner) {
       const metadataHeaders: Headers = {};
-      if (this.imdsV2SessionTokenUrl) {
+      // Only retrieve the IMDSv2 session token if both the security credentials and region are
+      // not retrievable through the environment.
```

Comment:
Perhaps another sentence as to why we have this here for future team to reference. Some questions we may ask:
- does this break something if we call it?
- does this improve performance?